pkg/cmd/render/render.go

```diff
@@ -227,15 +236,48 @@ func (r *renderOpts) Run() error {
 	return WriteFiles(&r.generic, &templateData.FileConfig, templateData)
 }
 
+// setBootstrapIP gets a list of IPs from local interfaces and returns the first IPv4 or IPv6 address.
+func (t *TemplateData) setBootstrapIP() error {
+	var bootstrapIP string
+	ips, err := ipAddrs()
+	if err != nil {
+		return err
+	}
+
+	for _, addr := range ips {
+		ip := net.ParseIP(addr)
+
+		// IPv6
+		if t.SingleStackIPv6 && ip.To4() == nil {
+			bootstrapIP = addr
+			break
+		}
+		// IPv4
+		if !t.SingleStackIPv6 && ip.To4() != nil {
+			bootstrapIP = addr
+			break
+		}
+	}
+
+	if bootstrapIP == "" {
+		return fmt.Errorf("no IP address found for bootstrap node")
+	}
+
+	t.BootstrapIP = bootstrapIP
+	return nil
+}
+
 func (t *TemplateData) setEtcdAddress() {
 	// IPv4
 	allAddresses := "0.0.0.0"
 	localhost := "127.0.0.1"
+	bootstrapIP := t.BootstrapIP
 
 	// IPv6
 	if t.SingleStackIPv6 {
-		allAddresses = "[::]"
+		allAddresses = "::"
 		localhost = "[::1]"
+		bootstrapIP = "[" + t.BootstrapIP + "]"
```

Comment:
Name it something else if it is escaped

Reply:
host-etcd-2 should have an unescaped value

Reply:
yeah got too late I will step through this today,